vulkan_generator/vulkan_parser/defines_parser.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This module is responsible for parsing Vulkan defines
+
+    It parses all defines except some of the defines that are defined under enums as "API Constants"
+"""
+
+import xml.etree.ElementTree as ET
+
+from vulkan_generator.vulkan_parser import types
+
+
+def parse_define_by_attribute(define_elem: ET.Element) -> types.VulkanDefine:
+    """Parses a define that has a name attribute
+
+    If a define has name attribute, it's usually a preprocessor decision logic
+    """
+    name = define_elem.attrib["name"]
+    value = define_elem.text
+
+    return types.VulkanDefine(
+        variable_name=name,
+        value=value
+    )
+
+
+def parse_define_by_tag(define_elem: ET.Element) -> types.VulkanDefine:
+    """Parses a define that has a name tag
+
+    If a define has a name tag, it's "usually" a macro
```

Comment:
Can we give better definition of "usually" here?

Reply:
I wish I could... but I can't it's the XML's irregularity. It can be a macro or sometime a constant value. There is no rule. But in our actual code we should not need this. This is only required for Vulkan.h.